data-protocols/ids/ids-api-multipart-dispatcher-v1/src/main/java/org/eclipse/dataspaceconnector/ids/api/multipart/client/sender/MultipartArtifactRequestSender.java

```
@@ -0,0 +1,102 @@
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.eclipse.dataspaceconnector.spi.iam.IdentityService;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.DataRequest;
+
+import java.net.URI;
+
+public class MultipartArtifactRequestSender extends IdsMultipartSender<DataRequest, MultipartRequestInProcessResponse> {
+
+    private final TransformerRegistry transformerRegistry;
+
+    public MultipartArtifactRequestSender(String connectorId,
+                                          OkHttpClient httpClient,
+                                          ObjectMapper objectMapper,
+                                          Monitor monitor,
+                                          IdentityService identityService,
+                                          TransformerRegistry transformerRegistry) {
+        super(connectorId, httpClient, objectMapper, monitor, identityService);
+        this.transformerRegistry = transformerRegistry;
+    }
+
+    @Override
+    public Class<DataRequest> messageType() {
+        return DataRequest.class;
+    }
+
+    @Override
+    protected String getConnectorId(DataRequest request) {
+        return request.getConnectorId();
+    }
+
+    @Override
+    protected String getConnectorAddress(DataRequest request) {
+        return request.getConnectorAddress();
+    }
+
+    @Override
+    protected Message buildMessageHeader(DataRequest request, DynamicAttributeToken token) throws Exception {
+        var asset = request.getAsset();
+        IdsId id = IdsId.Builder.newInstance()
+                .value(asset.getId())
+                .type(IdsType.ARTIFACT)
+                .build();
+        var transformationResult = transformerRegistry.transform(id, URI.class);
+        if (transformationResult.hasProblems()) {
+            throw new EdcException("Failed to create artifact ID from asset.");
```

Not to change now (can be done in a later PR), but can we report the errors and return an error message to the client?